server/conn.go

```diff
@@ -391,16 +392,17 @@ func (cc *clientConn) openSessionAndDoAuth(authData []byte) error {
 	if err != nil {
 		return errors.Trace(err)
 	}
-	if !cc.server.skipAuth() {
-		// Do Auth.
+	host := variable.ServerHostname
```

Comment:
it seems mysql here will got "localhost" and do auth logic

Reply:
Yes, I know it. I used to set host to a const string here, due to I have not checked the implementation of MySQL. If it is critical, I'll check when will MySQL set "localhost" and whether there are other cases.

Reply:
I think a const is valid here :-) MySQL has a special handling/meaning of "localhost".  Clients that connect to "localhost" will be redirected to a socket etc.